### PR TITLE
Allow customization of maxAliasesForCollections loader option

### PR DESCRIPTION
**Pull request by hcdeng**
This addresses https://github.com/circe/circe-yaml/issues/136.

SnakeYaml has maxAliasesForCollections defaulted to 50 in its LoaderOptions. This PR enables customization of this limit, following a similar pattern as [Printer](https://github.com/circe/circe-yaml/blob/master/src/main/scala/io/circe/yaml/printer/package.scala), while retaining the current API.

### CODE REVIEW

File: src/main/scala/io/circe/yaml/Parser.scala
Comment (travisbrown) [inline src/main/scala/io/circe/yaml/Parser.scala:132]: Kind of nitpicky, but can we rename this to `default` (which follows the naming convention in e.g. circe-generic-extras) and put an explicit type annotation on it?

**Review by travisbrown — COMMENTED** [review]
Thanks! This looks good to me but ideally we'd also have a sign-off from one of the more core circe-yaml maintainers.

File: src/main/scala/io/circe/yaml/Parser.scala
Comment (hcdeng) [reply src/main/scala/io/circe/yaml/Parser.scala]: Yup, updated

**Comment by hcdeng** [comment]
Thanks @travisbrown ! Would that be @jeffmay or someone else?

**Review by jeffmay — APPROVED** [review]

**Comment by jeffmay** [comment]
looks like some more scalafmt issues...

**Comment by hcdeng** [comment]
thanks - I think I got them all this time. @jeffmay do you mind kicking off ci again?

**Comment by hcdeng** [comment]
Hi @jeffmay or @travisbrown - can one of you please run CI again? Unfortunately I can't as a first-time contributor.

**Comment by hcdeng** [comment]
Thanks! Looks ready to merge when you get the chance.

**Comment by travisbrown** [comment]
I'm publishing an 0.14.1 release right now, but will merge this for the upcoming 0.15 series after that.

**Comment by Ushhud** [comment]
Hi @travisbrown and @jeffmay, would you be able to merge @hcdeng's changes given that the checks have passed? If so, would you happen to have an ETA on the version and its release?

Thank you!

**Comment by travisbrown** [comment]
@Ushhud I am no longer contributing to any Scala open source projects, for the reasons outlined [here](https://meta.plasm.us/posts/2021/11/17/scala-open-source/), but the current Circe maintainers may be able to help.

**Comment by sbrnunes** [comment]
Hi @jeffmay, we just ran across this issue as well. Do you have any ETA to release 0.15? I see you merged this to master on September 14, would be great to get this issue released whenever you get the chance. Thank you. 🙇

**Pull request merged**
